pageserver/src/tenant/timeline.rs

```diff
@@ -1559,14 +1565,17 @@ impl Timeline {
     /// Obtains a temporary lease blocking garbage collection for the given LSN
     pub(crate) fn make_lsn_lease(
         &self,
-        _lsn: Lsn,
+        lsn: Lsn,
         _ctx: &RequestContext,
     ) -> anyhow::Result<LsnLease> {
-        const LEASE_LENGTH: Duration = Duration::from_secs(5 * 60);
         let lease = LsnLease {
-            valid_until: SystemTime::now() + LEASE_LENGTH,
+            valid_until: SystemTime::now() + LsnLease::DEFAULT_LENGTH,
         };
-        // TODO: dummy implementation
+
+        {
+            let mut gc_info = self.gc_info.write().unwrap();
+            gc_info.leases.insert(lsn, lease.clone());
```

Comment:
To make this robust against some future change where the lease length is not always the same, we should only insert a new lease if the valid_until is greater than the lease that's already in `gc_info.leases` for the same LSN

Reply:
addressed using the entry API.